provider-naming-standardisation/README.md

```diff
@@ -0,0 +1,76 @@
+---
+creation_date: 2024-02-12
+issues:
+- https://github.com/giantswarm/giantswarm/issues/29790
+owners:
+- https://github.com/orgs/giantswarm/teams/team-horizon
+- https://github.com/orgs/giantswarm/teams/team-phoenix
+- https://github.com/orgs/giantswarm/teams/team-rocket
+state: review
+summary: Giant Swarm cluster providers should use a consistent naming scheme across all platforms.
+---
+
+# Standardisation of provider naming scheme
+
+## Current situation
+
+Depending on the provider, we use a different naming scheme. Our current provider names are:
+
+- CAPA
+- CAPZ
+- cloud-director
+- gcp
+- openstack
+- vsphere
+
+For CAPA/CAPZ, this is currently unavoidable as the aws/azure names refer to our vintage providers. However, this blocker is removed once all vintage clusters have been migrated to CAPI.
+
+Having this split in naming conventions causes some confusion, and whilst this is relatively minor it makes sense to pick a specific scheme (especially before adding any more providers). There appears to be no documented reason for choosing either scheme (aside from the aforementioned collision with our vintage providers) - this RFC aims to address that.
+
+Note: a _platform_ as used in this document can be thought of as a _location_ (e.g. I deployed on AWS). Whereas a _provider_ is the mechanism used to deploy (e.g. I deployed on AWS using CAPA).
+
+## Moving forward
+
+There are two options moving forward. One is to use the CAPx acronym, and the other is to use the platform name.
+
+### CAPx
+
+Pros:
+
+- Caters for multiple providers per platform (e.g. CAPA and EKS clusters on AWS).
+- More uniform than platform names.
+- Widely used and understood in the Kubernetes industry.
+- Unique and unambiguous.
+
+Cons:
+
+- Not as descriptive and harder to grok at a glance.
+
+### Platform name
+
+Pros:
+
+- Descriptive and easy to understand.
+
+Cons:
+
+- Creates a 1:1 mapping between provider and plaform (see CAPA vs EKS above).
+
+## Outcome
+
+I would suggest that given the possibility of having multiple providers for a single platform, we should rename gcp/vcd/vsphere/openstack in order to align with the CAPx naming scheme.
+
+### Changes
```

Comment:
If we decide for this, I think we should at the same time decide how to distinguish sub-providers (CAPA EC2 vs. CAPA EKS, CAPZ with VMs vs. later on maybe CAPZ with AKS, ...)

Reply:
> The higher level would be AWS, Azure, Google etc., and maybe something like "on-prem". 

I would stay away from cloud vs on-prem. At the moment the on-prem providers are vcd and vsphere but in the future we might have bare-metal, nutanix or who knows what, which will be very different (like capa vs capv different).

- AWS
  - CAPI (or CAPA)
  - Vintage
  - EKS
- Azure
  - CAPI (or CAPZ)
  - AKS
- vSphere (only 1 flavour)
- Cloud Director (only 1 flavour)
- ...

But that would maybe require quite a significant refactor in the operators to support 2 properties instead of one.

@AndiDog The latest direction is towards the original plan as per https://github.com/giantswarm/rfc/pull/86#issuecomment-1966295463

Reply:
My 2c here :) 

Big disclaimer: assuming it's future and we just have CAPI, because I don't think we should do any changes to try to make it nicer for vintage+CAPI mix 🙈 

IMO we are trying to put multiple properties in to a provider name here.

To take a few steps back, we have these providers in CAPI product ("providers" as actual names of cloud or on-prem providers like AWS and Azure, not names of projects like CAPA and CAPZ):
- AWS
- Azure
- vSphere
- Cloud Director

Then some of these above have 2 "flavours" (not sure if there is a better single word), where one "flavour" is the one where control plane is deployed by kubeadm and CAPA controllers, and the other one is where the control plane is fully managed by the provider itself. So above expanded would be:
- AWS
  - Self-managed or CAPI-managed control plane
  - Provider-managed control plane (EKS)
- Azure
  - Self-managed or CAPI-managed control plane
  - Provider-managed control plane (AKS)
- vSphere
- Cloud Director

To elaborate on the difference:
  - Self-managed or CAPI-managed control plane (kubeadm is doing the provisioning, CAPI+CAPx controllers manage the nodes)
  - Provider-managed control plane - cloud provider fully manages the control plane nodes, maybe you get to configure something but that's it, you don't deploy them and maybe you don't even see them.

So IMHO we should not hide the above difference behind a provider name, but we should introduce managed vs self-managed concept here, because that is essentially the difference.

So if needed in manifests somewhere, or if you have to specify the provider name (e.g. kubectl-gs and other tooling), we could have something like this and you specify the other info as well:
```
provider: aws
managed: true/false
---
provider: azure
managed: true/false
---
provider: vsphere
---
provider: cloud-director
```

Then we might even have to expand the above with `adopted: true/false` or `imported: true/false` (for the clusters that we have not deployed initially, but we have adopted/imported them with CrossPlane I think).

With the above in mind, we could think how to name components, so we could maybe have:
- `cluster-aws` + `cluster-eks` (or `cluster-eks` could be `cluster-aws-managed` but I don't think it matters)
- `aws-app-collection` and `eks-app-collection`/`aws-managed-app-collection`/`managed-aws-app-collection` (or just a single `aws-app-collection` if it works for bot self-managed and managed)

Reply:
Even in upstream CAPI providers the names like CAPA and CAPZ are not provider names, and also them have AWS and Azure as provider names, so it's:
- cluster-api-provider-aws, 
- cluster-api-provider-azure,
- etc

This means that cluster-api-provider-aws (provider being AWS) covers both self-managed clusters (where kubeadm does the provisioning) and clusters managed by the cloud provider (where EKS manages the control plane).

IMO we should try to align our terminology with this, so the name of the provider is the same, but we have a new "parameter" that indicates if is it "self-managed" (kubeadm) or managed by the cloud-provider.